serverless.yml

```diff
@@ -122,6 +122,13 @@ functions:
               - DialogEventBatches
               - StreamArn
           startingPosition: LATEST
+          destinations:
+            onFailure:
+              arn:
+                Fn::GetAtt:
+                  - DistributeDialogEventFailureQueue
+                  - Arn
+              type: sqs
```

Comment:
So is this taking the event from the stream that caused the error and putting it on an SQS queue as well? What are the benefits of this/ how does it help the debugging process?

Reply:
If an event is unprocessable we want to put it somewhere rather than throw it away. We can also see the exact payload that was causing a problem in the SQS UI

Reply:
ah gotcha, so kinesis doesn't allow for easy inspection of items on the stream?
Also, does kinesis throw events away? I thought the point of a stream was that it persists events, unlike a queue which throws events away.

Reply:
"Throw away" is a bad term. The events are persisted in the stream for 7 days, _but_ the consumer pointer will move on after a certain number of retries

Reply:
Cool this makes sense. But I wonder if an even more helpful debugging step would be to add rollbar to all of our lambdas. Lmk if I'm mistaken, but I can't see any rollbar stuff in the dialog engine.

Reply:
That’s a good idea. Data dog is good, but we don’t get locals. Lets add that to the backlog

Reply:
cool I added it to the backlog